docs/guides/configurations.md

```diff
@@ -0,0 +1,403 @@
+By default, very high priority is set to agent pods in the Chart.
+
+
+#### Pod anti-affinity
+
+It is recommended to schedule agent pods on different nodes as much as possible.
+To achieve this, the following podAntiAffinity is set by default.
+
+```yaml
+agent:
+  affinity:
+    podAntiAffinity:
+      preferredDuringSchedulingIgnoredDuringExecution:
+        - weight: 100
+          podAffinityTerm:
+            topologyKey: kubernetes.io/hostname
+            labelSelector:
+              matchExpressions:
+                - key: app
+                  operator: In
+                  values:
+                    - vald-agent-ngt
+```
+
+### Gateway
+
+#### Ingress
+
+Ingress for gateway can be configured by `gateway.ingress` field object.
+It is important to set your host to `gateway.ingress.host` field.
+`gateway.ingress.servicePort` should be `grpc` or `rest`.
+
+If you're using Vald-Helm-operator, you can check the ingress host by using kubectl command.
+
+```bash
+$ kubectl get valdrelease
+NAME           INGRESS                  AGE
+vald-cluster   gateway.vald.vdaas.org   9d
+```
+
+#### Index replica
+
+`gateway.gateway_config.index_replica` means how many number of agent pods that a vector will be inserted.
+
+#### Discoverer request duration
+
+`gateway.gateway_config.discoverer.duration` means a frequency to ask agent pod's IPs to the discoverer.
+If discoverer's CPU utilization is too high, try to make this value longer or reduce the number of gateway pods.
+
+
+#### Meta cache
+
+Gateway has a cache functionality for meta data.
+It can be enabled by `gateway.gateway_config.meta.enable_cache` and the behaviors controlled by `gateway.gateway_config.meta.cache_expiration` and `gateway.gateway_config.meta.expired_cache_check_duration`.
+
+#### Resource requests and limits
+
+Gateway's resource requests and limits depend on the request traffic and available resources.
+If the request traffic varies largely, it is recommended to enable HPA for gateway and adjust the resource requests.
+
+#### Init containers
+
+Gateway should wait for discoverer, agent, meta, and compressor to be ready because it depends on these components.
+For this purpose, "wait-for" type initContainers are provided in the Chart.
+
+```yaml
+  initContainers:
+    - type: wait-for
+      name: wait-for-manager-compressor
+      target: compressor
+      image: busybox
+      sleepDuration: 2
+      ...
+```
+
+"wait-for" type initContainers check readiness port of the target component is ok or not every "sleepDuration" seconds.
+Once it became ready, the initContainer returns zero and become "Completed" status.
+
+The definitions can be found in `_helpers.tpl` in Chart's templates directory.
+
+
+### Discoverer
+
+#### Resource requests and limits
+
+The number of discoverer pods and resource limits are determined by the configurations of your gateways and index managers, because APIs of discoverers are called by gateways and index managers.
```

Comment:
```suggestion
The number of discoverer pods and resource limits are determined by the configurations of your gateways and index managers because APIs of discoverers are called by gateways and index managers.
```